pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3837,6 +3838,24 @@ public void endReplaceSegments(String tableNameWithType, String segmentLineageEn
         tableNameWithType, segmentLineageEntryId);
   }
 
+  /**
+   * Internal method to initiate pageCache warmup for a table before the new refresh segments are available
```

Comment:
Update the javadoc to something like:
This method can be overridden to perform operations before updating the routing table to switch routing from `segmentsFrom` to `segmentsTo`. One example usage would be ...

Reply:
Sure updated it like that